packages/page-statistics/src/provider/hooks/useAuthorities.ts

```diff
@@ -37,19 +50,390 @@ type AuthorityStats = {
+      callValidatorOfSession({
+        variables: {
+          sessionValidatorId: `${metaData.val.activeSession}-${authorityId}`,
+          validatorId: authorityId,
+        },
+      }).catch((e) => {
+        setStats({
+          val: null,
+          isFailed: true,
+          isLoading: false,
+          error: e.message,
+        });
+      });
+    }
+  }, [metaData, callValidatorOfSession, authorityId]);
+  useEffect(() => {
+    const subscription = queryKeyGen.observable
+      .map((res): AuthorityDetails['keyGens'] => {
+        if (res.data && res.data.sessionValidators) {
+          const sessionValidators = res.data.sessionValidators;
+          const items = sessionValidators.nodes.map((node): KeGenKeyListItem => {
+            const session = node?.session!;
+            const publicKey = session.publicKey!;
+            return {
+              id: publicKey.id,
+              session: session.id,
+              publicKey: publicKey.uncompressed!,
+              height: '0',
+              authority: String(session.sessionValidators.totalCount),
+            };
+          });
+          return {
+            isLoading: false,
+            isFailed: false,
+            val: {
+              items,
+              pageInfo: {
+                count: sessionValidators.totalCount,
+                hasPrevious: sessionValidators.pageInfo.hasPreviousPage,
+                hasNext: sessionValidators.pageInfo.hasNextPage,
+              },
+            },
+          };
+        }
+        return {
+          isLoading: res.loading,
+          isFailed: Boolean(res.error),
+          error: res.error?.message ?? undefined,
+          val: null,
+        };
+      })
+      .subscribe(setKeyGens);
+    return () => subscription.unsubscribe();
+  }, [queryKeyGen]);
+  useEffect(() => {
+    const subscription = queryValidatorOfSession.observable
+      .map((res): AuthorityDetails['stats'] => {
+        if (res.data && res.data.sessionValidator) {
+          const sessionValidator = res.data.sessionValidator;
+          const sessionValidators = res.data.sessionValidators;
+          const counter = sessionValidators?.aggregates?.distinctCount?.id as number;
+          const threshold = sessionValidator.session?.keyGenThreshold! as QueryThreshold;
+          const auth = mapSessionAuthValidatorNode(sessionValidator);
+          const stats: AuthorityStats = {
+            numberOfKeys: String(counter),
+            keyGenThreshold: {
+              val: String(threshold.current),
+              inTheSet: auth.isBest,
+            },
+            nextKeyGenThreshold: {
+              val: String(threshold.next),
+              inTheSet: auth.isNextBest,
+            },
+            pendingKeyGenThreshold: {
+              val: String(threshold.pending),
+              inTheSet: auth.isBest,
+            },
+            reputation: auth.reputation,
+            uptime: '100',
+          };
+          return {
+            error: '',
```

Comment:
```suggestion
            error: undefined,
```

Prefer setting error to `undefined` in success case rather than empty string. Here and other relevant places.